paper/src/main/java/vg/civcraft/mc/civmodcore/world/locations/chunkmeta/ChunkMetaFactory.java

```diff
@@ -1,36 +1,35 @@
 package vg.civcraft.mc.civmodcore.world.locations.chunkmeta;
 
-import java.util.Collection;
-import java.util.HashMap;
-import java.util.Map;
-import java.util.Map.Entry;
-import java.util.TreeMap;
+import java.util.*;
 import java.util.function.Supplier;
 
 public class ChunkMetaFactory {
 
 	private static ChunkMetaFactory instance;
+
 	public static ChunkMetaFactory getInstance() {
```

Comment:
I think we can leave it as is because **instance** field never changes during the life cycle.
It is guaranteed that **instance** will be created synchronously before it will start to be used **async**.
We first register a plugin in ChunkMetaAPI.java and only after that it will be started to use in threads:
```
		if (!storageEngine.stayLoaded()) {
			//if a plugin preloads all data, we don't want to do anything on chunk load/unload
			ChunkMetaFactory metaFactory = ChunkMetaFactory.getInstance();
			metaFactory.registerPlugin(plugin.getName(), id, (Supplier<ChunkMeta<?>>) (Supplier<?>) emptyChunkCreator);
			LoadStatisticManager.registerPlugin(plugin.getName(), id);
		}
```

Therefore I think we are safe here.